New 'Hello World' page

rafaelcamaram opened the pull request:
This PR aims basically to add a new 'Hello World' page as it was requested by #90 (created by @Rich-Harris).
 
[Figma File](https://www.figma.com/file/atJYebOMeqoS6MWzxjOfE4/svelte-template?node-id=0%3A1) by @vedam 

### Tests
- [X] Run the tests with `pnpm test` and lint the project with `pnpm lint`

### CLI - Preview - (first step)
<img width="911" alt="Screen Shot 2021-04-09 at 10 56 48" src="https://user-images.githubusercontent.com/9087886/114190968-4a55b200-9922-11eb-9928-30c6b81fd9ed.png">

### Default template - Preview
<img width="1680" alt="Screen Shot 2021-04-08 at 22 02 01" src="https://user-images.githubusercontent.com/9087886/114113933-23fa2d00-98b6-11eb-8200-303bee9d77b8.png">

<img width="1680" alt="Screen Shot 2021-04-08 at 22 01 44" src="https://user-images.githubusercontent.com/9087886/114113902-18a70180-98b6-11eb-8cac-e9bbdd89fc31.png">

### Skeleton template - Preview
<img width="1678" alt="Screen Shot 2021-04-08 at 22 33 26" src="https://user-images.githubusercontent.com/9087886/114116135-80f7e200-98ba-11eb-9ea3-2f0e5979fddf.png">

### Observation
I'm still performing my first baby steps with Svelte, so feel free to let me know if I can improve something 😛

benmccann commented:
Wow, cool! This PR if very much appreciated!

My main question would be around the skeleton vs hello world app. I don't have a fully formed opinion on this yet. Some things to think about:
* I wonder if it will become burdensome to maintain both
* Should we combine the hello world template with https://github.com/sveltejs/kit/tree/master/examples/svelte-kit-demo somehow?
* Will having both make things like [svelte-adders](https://github.com/svelte-add/svelte-adders#-sveltekit) harder? (CC @babichjacob)

Also, if you happen to want to do any more of this type of work you might be interested in https://github.com/sveltejs/kit/issues/244 as well :smile:

rafaelcamaram commented:
Hey @benmccann, I'm really excited to know that you appreciated it :) 

IMHO, I only applied the `skeleton` template because it was suggested originally on the issue but I believe we could follow what you mentioned keeping only the `default` one. If the user wants to simplify it and get the same files we have at the `skeleton` template, it'll be necessary to remove only 5 files (or use the `svelte-adder`, as you mentioned). What do you think? Can we remove it then? 

Oh, and definitely, I'll take a look at #244 and dedicate some time to it! 😉

benmccann commented on packages/create-svelte/template/src/lib/DarkModeToggle.svelte:
```diff
@@ -0,0 +1,44 @@
+            right: 32px;
+        }
+    }
+</style>
```
nit: it'd be good to add a blank line at the end of the file

benmccann commented on packages/create-svelte/template/src/lib/DarkModeToggle.svelte:
```diff
@@ -0,0 +1,44 @@
+<script>
+    import { afterUpdate } from 'svelte';
```
there's a mix of tab and space indentation here

benmccann commented on packages/create-svelte/template/src/routes/blog.svelte:
```diff
@@ -0,0 +1,5 @@
+	<title>Blog</title>
+</svelte:head>
+
+<h1>Blog page</h1>
```
nit: it'd be good to add a blank line at the end of the file

benmccann commented on packages/create-svelte/template-skeleton/.gitignore:
```diff
@@ -0,0 +1,5 @@
+node_modules
+/.svelte
+/build
+/functions
```
nit: it'd be good to add a blank line at the end of the file

benmccann commented on packages/create-svelte/template-additions/src/app.css:
```diff
@@ -1,4 +1,80 @@
+	flex-direction: column;
+}
+
+h1,
```
I personally find it a bit uncommon to have each of these on their own line compared to `h1, h2, p` on one line (same below)

benmccann commented:
Yeah, makes sense. It's probably a good idea to have both. The question in my mind would mostly be around how to accomplish it. Right now there's two copies of a lot of files like https://github.com/rafaelcamaram/kit/blob/camara/gh-90/packages/create-svelte/template-skeleton/README.md and we'd have to modify two copies of them everytime we want to make a change and I imagine they'd get out of sync and might become annoying to deal with

I'd want the other maintainers to weigh in on these questions, but I personally might suggest the following:
* Just have the one template, but make the script strip out all the stuff we don't want to create the skeleton. Add a test to make sure that script succeeds
* Remove the SvelteKit demo project in favor of this new project. The demo project looks really outdated by comparison now and doesn't provide anything additional
* Remove the CSS option  and make it purely a svelte-adder thing. We can make `create-svelte` print a message pointing to `svelte-adders`. I doesn't make sense to me that `create-svelte` is responsible for creating Less and Sass templates, but not PostCSS/Tailwind which are probably more widely used in the Svelte community

Thanks again for all your help with this!

rafaelcamaram commented:
Hey @benmccann, I just applied all the suggestions you gave. Please, let me know if I misunderstood something or if there is something more to improve :) 

- [X] Created a better way to allow the user to choose between both templates but using only a single one -- using modifications in order to build the `skeleton` one;
- [X] Removed the SvelteKit demo project -- do we need to update the docs as well?
- [X] Removed the support for CSS modifications from the CLI;
- [X] Applied the changes requested on PR comments;

dummdidumm commented on packages/create-svelte/cli/index.js:
```diff
@@ -77,9 +77,14 @@ async function main() {
+	console.log(
+		`\nVisit ${bold(
+			cyan('https://github.com/svelte-add/svelte-adders')
+		)} and unleash the possibilities with a community project of commands to add particular functionalities to Svelte projects.\n`
```
This reads a little too non-neutral for a CLI setup for me 😄 not sure what others think about that

pngwn replied:
Yes, I agree. I'd like to see how the `svelte-adders` project develops before we make it an official recommendation.

babichjacob replied:
It's already in the CLI and it's fortunately very easy to miss because I'm not interested in having a large amount of attention (yet).

babichjacob replied:
It's a community-made and also really buggy code modification tool, so everywhere it's linked to should be removed or replaced with a link to the [official integrations repo](https://github.com/sveltejs/integrations).

rafaelcamaram replied:
@dummdidumm @babichjacob Oh I see, it makes sense :) What do you recommend then? Replace the svelte-adders link with the official integrations repo one?

babichjacob replied:
Sure, that's good.

rafaelcamaram replied:
Done :) 

<img width="764" alt="Screen Shot 2021-04-12 at 13 36 35" src="https://user-images.githubusercontent.com/9087886/114429761-1bdf0d80-9b94-11eb-827e-e17998813d02.png">

dummdidumm commented on packages/create-svelte/cli/index.js:
```diff
@@ -103,25 +108,25 @@ async function main() {
 	});
-	await add_typescript(target, ts_response.value);
-
-	const css_response = await prompts({
```
Personally I would have left that in it's really easy to setup, we don't even need to add much code modifications. Also, do we have any data how many people would choose what? I think people would opt for scss quite a bit just because many component libraries use it for their styling/theming.

rafaelcamaram replied:
Hm, I see your point.. The only drawback that I see is that we would need to maintain 3 different styling files (I believe that's why @benmccann suggested removing it) but i's not a big deal for me -- it'd be easy to revert it.

@dummdidumm @benmccann How I can move forward with it?

benmccann replied:
We chatted about this part for a bit and it sounds like @Rich-Harris and @pngwn also vote to take out the CSS stuff

dummdidumm commented on packages/create-svelte/template/src/app.css:
```diff
@@ -0,0 +1,80 @@
+	--background-without-opacity: rgba(255, 255, 255, 0.7);
+}
+
+[data-theme='dark'] {
```
This applies to all styles with that attribute. Should we scope it to `root`?

dummdidumm commented on packages/create-svelte/template/src/lib/Counter.svelte:
```diff
@@ -1,36 +1,143 @@
+		opacity: 0.3;
+	}
+
+	.counterControls button {
```
These are camelCase, below it's snake-case. We should consolidate that

benmccann commented on packages/create-svelte/template/src/lib/HeaderNavigation.svelte:
```diff
@@ -0,0 +1,73 @@
+		background-color: var(--accent-color);
+		transform: rotate(45deg);
+	}
+</style>
```
it'd probably be nice to add a new line to the end of this file

benmccann commented on packages/create-svelte/template/static/moon-icon.svg:
```diff
@@ -0,0 +1,3 @@
+<svg width="22" height="22" viewBox="0 0 22 22" fill="none" xmlns="http://www.w3.org/2000/svg">
+    <path d="M13.0264 1.20543C10.2914 1.91848 8.27275 4.4053 8.27275 7.36365C8.27275 10.8782 11.1218 13.7273 14.6363 13.7273C17.5947 13.7273 20.0815 11.7086 20.7945 8.97365C20.9292 9.628 21 10.3058 21 11C21 16.5229 16.5229 21 11 21C5.47715 21 1 16.5229 1 11C1 5.47715 5.47715 1 11 1C11.6942 1 12.372 1.07075 13.0264 1.20543Z" stroke="#676778" stroke-width="2" stroke-linejoin="round"/>
+</svg>
```
it'd probably be nice to add a new line to the end of this file

benmccann commented on packages/create-svelte/template/static/sun-icon.svg:
```diff
@@ -0,0 +1,12 @@
+    <path d="M4.75 20.5C5.44035 20.5 6 19.9404 6 19.25C6 18.5597 5.44035 18 4.75 18C4.05964 18 3.5 18.5597 3.5 19.25C3.5 19.9404 4.05964 20.5 4.75 20.5Z" fill="white" fill-opacity="0.6"/>
+    <path d="M1.75 13.25C2.44036 13.25 3 12.6904 3 12C3 11.3097 2.44036 10.75 1.75 10.75C1.05964 10.75 0.5 11.3097 0.5 12C0.5 12.6904 1.05964 13.25 1.75 13.25Z" fill="white" fill-opacity="0.6"/>
+    <path d="M4.75 6C5.44035 6 6 5.44035 6 4.75C6 4.05964 5.44035 3.5 4.75 3.5C4.05964 3.5 3.5 4.05964 3.5 4.75C3.5 5.44035 4.05964 6 4.75 6Z" fill="white" fill-opacity="0.6"/>
+</svg>
```
it'd probably be nice to add a new line to the end of this file

benmccann commented on packages/create-svelte/cli/modifications/add_typescript.js:
```diff
@@ -13,19 +13,37 @@ import {
+ * @param {boolean} is_skeleton_template
  */
-export default async function add_typescript(cwd, yes) {
+export default async function add_typescript(cwd, yes, is_skeleton_template) {
```
there was a question about whether we could reverse this to start with the typescript template and make it `remove_typescript`. we have `svelte-preprocess` which should already offer the ability to strip out the typescript stuff and would make it so that we don't have to update this script at all when we change the template. using `svelte-preprocess` might be a big change to include in this PR, but it might be a step towards that to at least reverse the logic even if still done manually

dummdidumm commented:
I think we should hold off on going further here until #989 or something similar is merged and rebuild upon that, else merge conflicts galore.

rafaelcamaram commented:
Since we got several conflicts here, I thought it would be easier to create a new PR based on the new `master`. 

New PR related to the new 'Hello World' templates: https://github.com/sveltejs/kit/pull/1007
